onnx/backend/test/case/node/dropout.py

```diff
@@ -12,21 +12,25 @@
 from onnx import helper
 
 
-def dropout(x, ratio=0, seed=0):  # type: ignore
-    np.random.seed(seed)
-    total_nb_indices = x.size
-    nb_to_dropout = total_nb_indices * ratio
-    indices_to_dropout = np.random.choice(total_nb_indices, int(nb_to_dropout), replace=False)
-    flattened_x = x.flatten()
+def dropout(X, drop_probability=0.5, seed=0, training_mode=False, return_mask=False):  # type: ignore
+    if drop_probability == 0 or training_mode is False:
+        if return_mask is True:
+            return X, np.ones(X.shape, dtype=bool)
+        else:
+            return X
 
-    for index in indices_to_dropout:
-        flattened_x[index] = 0.
-    y = flattened_x.reshape(x.shape)
-    return y
+    np.random.seed(seed)
+    mask = np.random.uniform(0, 1.0, X.shape) >= drop_probability
```

Comment:
We cannot match the output value in the onnx test, right? as the mask is randomly generated, and different backend will generate different mask.

Reply:
It is currently failing but as discussed offline I want to see it I can some how approximate the comparison but if that is not feasible I’ll skip _expect_ as you suggested.__

Reply:
After setting the seed attribute of dropout to zero and also in the reference implementation, tests on ORT pass.

Reply:
I have updated ORT test infra to count for number of zeros and make sure they are around the same count as baseline.